scripts/qiita-auto-processing

```diff
@@ -194,23 +217,33 @@ for pipeline in full_pipelines:
     #          of Step 4 but for debugging it makes sense to separate
     artifact_process = []
     children_compliant = []
+    cmd = commands[0]
     for a in artifacts_compliant:
-        cmd = commands[0]
         # getting all jobs, includen hiddens, in case the job failed
         jobs = a.jobs(cmd=cmd['command'], show_hidden=True)
-        params = [j.parameters.values for j in jobs]
+        params = [{k: str(v) for k, v in j.parameters.values.items()
+                   if k not in cmd['ignore_parameters']} for j in jobs]
 
         # checking that all required parameters of this command exist
         missing_parameters = []
         for p in cmd['parameters']:
             p = p['values']
             p.update({cmd['input_name']: str(a.id)})
-            if p not in params:
+            p_to_compare = p.copy()
+            for k in cmd['ignore_parameters']:
+                del p_to_compare[k]
+            if p_to_compare not in params:
                 missing_parameters.append(p)
             else:
                 for c in a.children:
-                    if c.processing_parameters.values == p:
-                        children_compliant.append(c)
+                    cpp = c.processing_parameters
+                    if cpp.command.name == cmd['command-name']:
+                        cparams = {
+                            k: str(v)
+                            for k, v in cpp.values.items()
+                            if k not in cmd['ignore_parameters']}
```

Comment:
This list comprehension pattern is repeated a couple times, any chance this can be put in a single function, or something like that?